### PR TITLE
Use Integer/Long compress and expand.

### PR DESCRIPTION
**Pull request by PaulSandoz**
Use the Integer/Long compress and expand implementations.

Fix compilation of perf tests for constant shift operations (brought in via merge that does not have perf tests).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Sandhya Viswanathan](https://openjdk.java.net/census#sviswanathan) (@sviswa7 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-vector pull/194/head:pull/194` \
`$ git checkout pull/194`

Update a local copy of the PR: \
`$ git checkout pull/194` \
`$ git pull https://git.openjdk.java.net/panama-vector pull/194/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 194`

View PR using the GUI difftool: \
`$ git pr show -t 194`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-vector/pull/194.diff">https://git.openjdk.java.net/panama-vector/pull/194.diff</a>

</details>

### CODE REVIEW

File: test/jdk/jdk/incubator/vector/templates/Perf-header.template [hunk 1]
```diff
@@ -77,6 +77,8 @@ public class $vectorbenchtype$ extends AbstractVectorBenchmark {
         return $Boxtype$.compare(a, ($type$) 0) != 0 ? a : b;
     }
 
+    private static final $type$ CONST_SHIFT = $Boxtype$.SIZE / 2;
+
```
Comment (sviswa7) [inline test/jdk/jdk/incubator/vector/templates/Perf-header.template:81]: This is only needed for integral types and not for float/double.

File: test/jdk/jdk/incubator/vector/templates/Perf-Scalar-header.template [hunk 2]
```diff
@@ -40,6 +40,8 @@ import org.openjdk.jmh.infra.Blackhole;
 public class $Type$Scalar extends AbstractVectorBenchmark {
     static final int INVOC_COUNT = 1; // To align with vector benchmarks.
 
+    private static final $type$ CONST_SHIFT = $Boxtype$.SIZE / 2;
+
```
Comment (sviswa7) [inline test/jdk/jdk/incubator/vector/templates/Perf-Scalar-header.template:44]: This is only needed for integral types and not for float/double.

**Review by sviswa7 — COMMENTED** [review]
I have only two minor comments. Rest of the patch looks good to me.

File: test/jdk/jdk/incubator/vector/templates/Perf-header.template (hunk 1 above)
Comment (PaulSandoz) [reply test/jdk/jdk/incubator/vector/templates/Perf-header.template]: Done

File: test/jdk/jdk/incubator/vector/templates/Perf-Scalar-header.template (hunk 2 above)
Comment (PaulSandoz) [reply test/jdk/jdk/incubator/vector/templates/Perf-Scalar-header.template]: Done. Also fixed in unit tests.

**Review by sviswa7 — APPROVED** [review]

**Pull request closed without merging**